core/eval.go

```diff
@@ -2287,3 +2287,33 @@ func evalLPOP(args []string, store *Store) []byte {
 
 	return Encode(x, false)
 }
+
+// GETSET atomically sets key to value and returns the old value stored at key.
+// Returns an error when key exists but does not hold a string value.
+// Any previous time to live associated with the key is
+// discarded on successful SET operation.
+//
+// Returns:
+// Bulk string reply: the old value stored at the key.
+// Nil reply: if the key does not exist.
+func evalGETSET(args []string, store *Store) []byte {
+	if len(args) != 2 {
+		return diceerrors.NewErrArity("GETSET")
+	}
+
+	var key, value = args[0], args[1]
+	getResp := evalGET([]string{key}, store)
+	// Check if it's an error resp from GET
+	if strings.HasPrefix(string(getResp), "-") {
+		return getResp
+	}
+
+	// Previous TTL needs to be reset
+	setResp := evalSET([]string{key, value}, store)
+	// Check if it's an error resp from SET
+	if strings.HasPrefix(string(setResp), "-") {
```

Comment:
Ideally this should be atomic, however, this will be a no-issue once we have the shared-nothing multi-threaded approach in-place. So ignoring this for now.